### PR TITLE
Add background processes

### LINKED ISSUE #24: Add background processes
We have the grammar and types for background processes (ones terminated with `&` instead of `;`), but no logic to actually run them asynchronously. `ptable.c` should already handle cleanup of the processes (via `waitpid`), so it's mostly about forking off here.

### PR DESCRIPTION
**Pull request by ammongit**
The syntactical plumbing for background processes has been around for a while, but not the actual ability to run them. This change adds three hooks into command execution: Builtins, normal commands, and chains.
In all three cases it does the following:
* forks off
* the parent calls `ptable_new`
* the parent jumps to the end, skipping over the actual process execution
* the child continues, executing the process
* when the child reaches the end, it calls `_exit`

Here's a drawing to help you visualize how it works:
(blue = parent, red = child)

**This changes the signature of the virtual `runnable_execute`** It adds a new bool argument called `background`, which does exactly what you'd think it does.

Fixes #24

### CODE REVIEW

**Comment by ammongit** [comment]
This does _not_ implement a way to disown children. Any children created by the shell will die with it.

**Comment by ammongit** [comment]
Fixes #14 and #27.

File: README.md [hunk 1]
```diff
@@ -50,4 +52,6 @@ Invoking the `Makefile` in the `src` directory will produce a debug build, which
 be created using the `debug` target when invoking `make`.
 
 ### Known Issues
-None.
+The process table sometimes leaves entries for children that have died.
```
Comment (BradleyCai) [inline README.md]: No longer true

File: src/builtins.c [hunk 2]
```diff
@@ -392,60 +643,116 @@ static int builtin_umask(int argc, const char **argv)
...
+			shell_error(argv[0], "invalid argument: %s", argv[1]);
+			return 1;
+		}
+	} else if (argc != 1) {
+		shell_error(NULL, "usage: %s [-s]", argv[0]);
+		return 1;
+	}
+	tty = ttyname(STDIN_FILENO);
+	if (verbose) {
+		if (tty) {
```
Comment (BradleyCai) [inline src/builtins.c]: Doesn't need braces

**Comment by ammongit** [comment]
Look into adding state tracking for all the processes in the table.

**Comment by ammongit** [comment]
Also look into adding the full command line into the process table's name (e.g. "{ echo hi; }" instead of "echo")

**Pull request merged**